spec/integration/opentracing_spec.rb

```diff
@@ -0,0 +1,249 @@
+
+        it { should be_an ElasticAPM::OpenTracing::Scope }
+        its(:elastic_span) { should be_a ElasticAPM::Transaction }
+
+        it 'is active' do
+          expect(::OpenTracing.active_span).to be subject.span
+        end
+      end
+    end
+
+    describe 'activation' do
+      it 'sets the span as active in scope' do
+        span = OpenTracing.start_span('name')
+        OpenTracing.scope_manager.activate(span)
+        expect(OpenTracing.active_span).to be span
+      end
+    end
+
+    describe '#inject' do
+      let(:context) do
+        ElasticAPM::TraceContext.parse(
+          '00-11111111111111111111111111111111-2222222222222222-00'
+        )
+      end
+      let(:carrier) { {} }
+
+      subject { ::OpenTracing.inject(context, format, carrier) }
+
+      context 'Rack' do
+        let(:format) { ::OpenTracing::FORMAT_RACK }
+
+        it 'sets a header' do
+          subject
+          expect(carrier['elastic-apm-traceparent']).to eq context.to_header
+        end
+      end
+
+      context 'Binary' do
+        let(:format) { ::OpenTracing::FORMAT_BINARY }
+
+        it 'warns about lack of support' do
+          expect(tracer).to receive(:warn).with(/Only injection via/)
+          subject
+        end
+      end
+    end
+
+    describe '#extract' do
+      let(:carrier) do
+        { 'HTTP_ELASTIC_APM_TRACEPARENT' =>
+          '00-11111111111111111111111111111111-2222222222222222-00' }
+      end
+
+      subject { ::OpenTracing.extract(format, carrier) }
+
+      context 'Rack' do
+        let(:format) { ::OpenTracing::FORMAT_RACK }
+
+        it 'returns a trace context' do
+          expect(subject).to be_a ElasticAPM::TraceContext
+          expect(subject.trace_id).to eq '11111111111111111111111111111111'
+        end
+      end
+
+      context 'Binary' do
+        let(:format) { ::OpenTracing::FORMAT_BINARY }
+
+        it 'warns about lack of support' do
+          expect(tracer).to receive(:warn).with(/Only extraction from/)
+          subject
+        end
+      end
+    end
+  end
+
+  describe 'example', :intercept do
+    before { ElasticAPM.start }
+    after { ElasticAPM.stop }
+
+    it 'traces nested spans' do
+      OpenTracing.start_active_span(
+        'operation_name',
+        tags: { test: '0' }
+      ) do |scope|
+        expect(scope).to be_a(ElasticAPM::OpenTracing::Scope)
+        expect(OpenTracing.active_span).to be scope.span
+        expect(OpenTracing.active_span).to be_a ElasticAPM::OpenTracing::Span
+
+        OpenTracing.start_active_span(
+          'nested',
+          tags: { test: '1' }
+        ) do |nested_scope|
+          expect(OpenTracing.active_span).to_not be_nil
+          expect(nested_scope.span).to eq OpenTracing.active_span
+
+          OpenTracing.start_active_span(
+            'namest',
+            ignore_active_scope: true
+          ) do |further_nested|
+            expect(further_nested).to_not be nested_scope
```

Comment:
I agree that this test doesn't make sense, as it indicates a false positive that `ignore_active_scope` is working as expected but it doesn't.